Ticket

```diff
@@ -0,0 +1,97 @@
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.util.*;
+import java.lang.*;
+
+public class Ticket extends TicketAbs implements Serializable {
+    Movie movie;  //associated movie with the ticket
+    Cinema cinema; //associated cinema of the movie, this will also provide seat number
+    Customer client;  //purchasing client details
+    String showtime;
+    int seatNo;   //in Cinema class, the free seat id will be passed to Ticket
+    double price;
+    String priceFileName;
+    
+    Scanner sc= new Scanner(System.in);
+
+    public Ticket(Movie a, Cinema b, Customer c, String d, int sit_no, String price_FileName) {
+    	movie = a;
+    	cinema = b;
+    	client = c;
+    	showtime = d;
+    	seatNo = sit_no;
+    	priceFileName = price_FileName;
+    	calculatePrice();
+    	
+    	
+    }
+    
+    public void calculatePrice() {
+    	//gets price from prices database and does algorithm on it.
+    	//prices database stores default values for this particular movie and cinema and showtime
+    	//if child or senior x0.8
+    	int[] price_list;
+    	int moviePrice;
+    	MovieType movieType;
+    	int clientAge;
+    	CinemaType cinemaType;
+    	int date;
+    	double multiplier = 1;
+    	
+    	price_list = getPricesFromDB();
+    	moviePrice = price_list[movie.getId()];
+    	
+    	movieType = movie.getType();
+    	clientAge = client.getAgeGroup();
+    	cinemaType = cinema.getType();
+    	if (showtime.compareTo("Sunday")==1||showtime.compareTo("Saturday")==1||
+    			showtime.compareTo("Holiday")==1) {
+    		multiplier = multiplier*1.5;
+    	}
+    	
+    	multiplier = multiplier * movieType * clientAge * cinemaType;
```

Comment:
The overall ideology for calculating prices looks ok, but the movieType isnt an int, its an enum of `MovieType` defined in `MovieType.java`, so you'll have to create a switch statement to account for the different cases. Same thing with `CinemaType` defined in `CInemaAbs.java`.